SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTest.cpp

```diff
@@ -128,29 +150,31 @@ void AEventTracingTest::GatherData()
 		return A.CreationTime > B.CreationTime;
 	});
 
-	bool bFoundClient = false;
-	bool bFoundWorker = false;
+	FPlatformProcess::Sleep(1); // Worker bug means file may not be flushed by the OS (WRK-2396)
+
+	int FoundClient = 0;
+	int FoundWorker = 0;
 	for (const FileCreationTime& FileCreation : FileCreationTimes)
 	{
-		if (!bFoundClient && FileCreation.FilePath.Contains("UnrealClient"))
+		if (FoundClient != 1 && FileCreation.FilePath.Contains("UnrealClient"))
 		{
 			GatherDataFromFile(FileCreation.FilePath);
-			bFoundClient = true;
+			FoundClient++;
 		}
 
-		if (!bFoundWorker && FileCreation.FilePath.Contains("UnrealWorker"))
+		if (FoundWorker != 2 && FileCreation.FilePath.Contains("UnrealWorker"))
 		{
 			GatherDataFromFile(FileCreation.FilePath);
-			bFoundWorker = true;
+			FoundWorker++;
 		}
 
-		if (bFoundClient && bFoundWorker)
+		if (FoundClient == 1 && FoundWorker == 2)
```

Comment:
Hmm maybe we make these numbers configurable to the tests? ClientsRequired = 1  WorkersRequired = 2